<launch>
  <arg name="odom_topic" />

  <!-- Convert odometry to Pose2D -->
  <node pkg="kobuki_node" type="getOdom2D.py" name="getOdom2D">
    <remap from="odom" to="$(arg odom_topic)" />
    <remap from="pose2d" to="/pose2d" />
  </node>

  <!-- Run main state machine -->
  <node pkg="follow_bot" type="control.py" name="control" output="screen">
    <remap from="joy" to="/joy"/>
    <remap from="odom" to="$(arg odom_topic)"/>
    <remap from="pose2d" to="/pose2d"/>
    <remap from="target" to="/target_pose"/>
    <remap from="cmd_vel" to="/cmd_vel_mux/input/teleop"/>
  </node>

  <!-- Node to estimate target position in odom space -->
  <node pkg="follow_bot" type="target_position_estimator.py" name="target_position_estimator" output="screen">
    <param name="angle_cutoff_offset" value="0.25"/>
    <remap from="target_pose" to="/target_pose"/>
    <remap from="target_point" to="/target_point"/>
    <remap from="pose2d" to="/pose2d"/>
    <remap from="scan" to="/scan"/>
  </node>

  <!-- Launch rviz with a premade config to show target, pointcloud, etc -->
  <node type="rviz" name="rviz" pkg="rviz" args="-d $(find follow_bot)/rviz_config.rviz"  />

  <!-- Listen for joystick -->
  <node type="joy_node" pkg="joy" name="joy_node">
    <remap from="joy" to="/joy" />
  </node>
</launch>
